cmd/nerdctl/run_network.go

```diff
@@ -164,12 +168,85 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if err := verifyContainerNetwork(cmd, netSlice); err != nil {
+			return nil, nil, "", nil, err
+		}
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+
+		walker := &containerwalker.ContainerWalker{
+			Client: client,
+			OnFound: func(ctx context.Context, found containerwalker.Found) error {
+				if found.MatchCount > 1 {
+					return fmt.Errorf("multiple containers found with prefix: %s", containerName)
+				}
+				containerID := found.Container.ID()
+				// share same hostname with the target container
+				lb, err := found.Container.Labels(ctx)
+				if err != nil {
+					return err
+				}
+				cmd.Flags().Set("hostname", lb[labels.Hostname])
```

Comment:
What is the use?

Reply:
for setting up the hostname, It's kinda confusing here, maybe should add a comment?
https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/run.go#L557-L573